src/components/events/SponsorGrid.tsx

```
@@ -134,34 +136,95 @@ export const SponsorGrid: React.FC<SponsorGridProps> = ({ eventId, isEditMode =
+          } else {
+            // No longer 100% visible, cancel timer
+            if (timerId) {
+              clearTimeout(timerId);
+              timerId = null;
+            }
+          }
+        });
+      },
+      {
+        threshold: 1.0, // Strict 100% visibility requirement
+      }
```

ESLint

[error] Insert `,`

(prettier/prettier)